doc/odoc_for_authors.mld

```diff
@@ -263,12 +263,79 @@ and
 + item 2
 + item 3
 v}
-For inline source code style, use square brackets: [ [ ... ] ]. For longer,
-preformatted sections of code, use the enclosing tags [ {[ .. ]} ].
-For verbatim (non-source) formatted sections, use the enclosing tags [{v ... v}].
 
+{3:code_blocks Code Blocks}
 
+There are various ways of inserting code elements in your documentation.
 
+{4:inline_code_blocks Inline Code Blocks}
+
+For inline, language agnostic source code style, use square brackets: [ [ ... ] ].
+
+{[
+(** Here, [f 0] is [None] *)
+]}
+
+All the other existing ways to insert code are meant to be used for code blocks,
+not inline code fragments. Those will be formatted as a code block and will end
+the paragraph. Attempting to use them inline will trigger a warning.
+
+{4:ocaml_code_blocks OCaml Code Blocks}
+
+OCaml code blocks can be written using the enclosing tags [ {[ ... ]} ].
+The code inside these blocks will be properly styled as OCaml in the generated
+doc.
+
+{v
+(** This how one binds a variable in OCaml:
+    {[
+    let x = 0
+    ]}
+*)
+v}
+
+{4:language_headers Language Headers}
+
+As of [odoc.2.0] it is possible to write blocks with explicit language headers,
+similar to markdown code blocks:
+
+{@markdown[
+Here is some python code:
+```python
+def f():
+  return 0
+```
+]}
+
+These blocks can be written using the enclosing tags [ {@<language>[ ... ]} ].
+The content of those block should be properly styled if odoc supports the
```

Comment:
That's not implemented yet. As soon as we implement this, we'll have to release odoc-parser and add a version constraint on Odoc, fixing my previous comment.